README.md

```diff
@@ -1,120 +1,240 @@
+
+If [`StableHashTraits.is_ordered`](https://beacon-biosignals.github.io/StableHashTraits.jl/stable/api/#StableHashTraits.is_ordered) returns `false` the elements are first `sort`ed according to [`StableHashTraits.hash_sort_by`](https://beacon-biosignals.github.io/StableHashTraits.jl/stable/api/#StableHashTraits.hash_sort_by).
+
+To hash the type, the string `"StructTypes.ArrayTYpe"` is hashed (ergo, the kind of array won't matter), and the type of the `elype` is hashed, according to its `StructType`. If the type `<: AbstractArray`, the `ndims(T)` is hashed if it is available.
+
+### `StructTypes.DictType`
+
+To hash the value, each key-value pair of a dict type is hashed, as returned by `StructType.keyvaluepairs(x)`.
+
+If [`StableHashTraits.is_ordered`](https://beacon-biosignals.github.io/StableHashTraits.jl/stable/api/#StableHashTraits.is_ordered) returns `false` the pairs are first `sort`ed according their keys using [`StableHashTraits.hash_sort_by`](https://beacon-biosignals.github.io/StableHashTraits.jl/stable/api/#StableHashTraits.hash_sort_by).
+
+To hash the type `"StructTypes.DictType"` is hashed (ergo, the kind of dictionary won't matter), and the type of the `keytype` and `valtype` is hashed, according to its `StructType`.
+
+### `AbstractRange`
+
+`AbbstractRange` constitutes an exception to the rule that we use `StructType`: for efficient hashing, ranges are treated as another first-class container object, separate from array types.
+
+The value is hashed as `(first(x), step(x), last(x))`.
+
+The type is hashed as `"Base.AbstractRange"` along with the type of the `eltype`, according to its `StructType`. Ergo, the type of range doesn't matter (just that it is a range).
+
+### `StructTypes{Number/String/Bool}Type`
+
+To hash the value, the result of `Base.write`ing the object is hashed.
+
+To hash the type, the value of `module_nameof_string(StructType))` is used. [`module_nameof_string`](https://beacon-biosignals.github.io/StableHashTraits.jl/stable/api/#StableHashTraits.parentmodule_name) can be used to provide a relatively stable name for an object composed of `string(parentmodule(T))` and `string(nameof(T))`, along with some additional regularizations and validations.
+
+### `StructType.CustomStruct`
+
+For any `StructType.CustomStruct`, the object is first `StructType.lower`ed and the result is hashed according to the lowered `StructType`.
+
+### `missing` and `nothing`
+
+There is no value hashed for `missing` or `nothing`; the type is hashed as the string `"Base.Missing"` and `"Base.Nothing"` respectively.
+
+### `StructType.{Null/Singleton}Type`
+
+These types error by default. You can opt in to behavior similar to `missing` and `nothing` by defining an appropriate [`transform_type`](https://beacon-biosignals.github.io/StableHashTraits.jl/stable/api/#StableHashTraits.type_identifier).
```

Comment:
Alternative: use `nameof_string` for the type. This should be quite stable (since we're not relying on the module)